### PR TITLE
[Stocks App] NNBD migration, refresh gen-l10n

### PR DESCRIPTION
**Pull request by shihaohong**
- migrate the stocks app to nnbd
- ran `flutter gen-l10n`
- ignores manually removed until https://github.com/flutter/flutter/pull/78975/ lands.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart [hunk 1]
```diff
@@ -126,22 +135,27 @@ class _StockStringsDelegate extends LocalizationsDelegate<StockStrings> {
...
 
-  // Lookup logic when language+country codes are specified.
-  switch (locale.languageCode) {
-    case 'en': {
-      switch (locale.countryCode) {
-        case 'US': return StockStringsEnUs();
-      }
-      break;
-    }
+// Lookup logic when language+country codes are specified.
```
Comment (goderbauer) [inline dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart:138]: nit: indentation

File: dev/benchmarks/test_apps/stocks/l10n.yaml [hunk 2]
```diff
@@ -16,3 +16,4 @@ output-localization-file: stock_strings.dart
 ## generating Flutter's localization files.
 synthetic-package: false
 template-arb-file: stocks_en.arb
+nullable-getter: false
```
Comment (goderbauer) [inline dev/benchmarks/test_apps/stocks/l10n.yaml:19]: almost all other options here are documented. Should this one get some documentation as well?

File: dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart [hunk 3]
```diff
@@ -126,22 +135,27 @@ class _StockStringsDelegate extends LocalizationsDelegate<StockStrings> {
...
+  break;
+}
+}
 
-  // Lookup logic when only language code is specified.
-  switch (locale.languageCode) {
-    case 'en': return StockStringsEn();
+// Lookup logic when only language code is specified.
+switch (locale.languageCode) {
+  case 'en': return StockStringsEn();
```
Comment (goderbauer) [inline dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart:150]: indentation?

File: dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings_en.dart [hunk 4]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+
+
```
Comment (goderbauer) [inline dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings_en.dart:6]: remove empty lines?

File: dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings_es.dart [hunk 5]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+
+
```
Comment (goderbauer) [inline dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings_es.dart:6]: remove?

File: dev/benchmarks/test_apps/stocks/lib/stock_data.dart [hunk 6]
```diff
@@ -71,12 +71,12 @@ class StockData extends ChangeNotifier {
...
       'domokit.github.io', 'examples/stocks/data/stock_data_$chunk.json');
 
-  http.Client _httpClient;
+  http.Client? _httpClient;
 
   static bool actuallyFetchData = true;
 
   void _fetchNextChunk() {
-    _httpClient.get(_urlToFetch(_nextChunk++)).then<void>((http.Response response) {
+    _httpClient?.get(_urlToFetch(_nextChunk++)).then<void>((http.Response response) {
```
Comment (goderbauer) [inline dev/benchmarks/test_apps/stocks/lib/stock_data.dart:79]: Isn't it an error if this is null and we were told to fetch the next chunk?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by shihaohong** [comment]
All the `stock_strings*.dart` files are all generated code, so the styling fixes would need to be made in the localizations tool itself. I added a comment in the header for the files to clarify this.

**Pull request merged**